compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -20,12 +20,14 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
     /// Report an error that a generic argument did not match the generic parameter that was
     /// expected.
     fn generic_arg_mismatch_err(
-        sess: &Session,
+        tcx: TyCtxt<'_>,
         arg: &GenericArg<'_>,
+        param: Option<&GenericParamDef>,
         kind: &'static str,
```

Comment:
```suggestion
        param: &GenericParamDef,
```
we can always pass in a `GenericParamDef` by storing both the arg and the param in `force_infer_lt`.
This should be cleaner than redundantly passing both the param def and it's kind str.

After this r=me